src/components/IKUpload/index.tsx

```diff
@@ -18,15 +18,18 @@ const IKUpload = forwardRef<HTMLInputElement, IKUploadProps & IKContextBaseProps
   const contextOptions = useContext(ImageKitContext);
   const { getIKClient } = useImageKitComponent({ ...props });
 
-  const abort = () => {
-    if (state.xhr) {
-      state.xhr.abort();
-    }
-  }
 
-  if (ref && (ref as any).current) {
-    (ref as any).current = { ...(ref as any).current, abort }
-  }
+  useEffect(() => {
```

Comment:
Why do we need useEffect?

Reply:
`useEffect` was used to resolve the bug with the custom upload button ref was not working after uploading one time.